### PR TITLE
ECR support for Compose style CI

### PR DESCRIPTION
**Pull request by CommanderK5**

### CODE REVIEW

File: pkg/executors/docker_compose_executor.go [hunk 1]
```diff
@@ -177,6 +177,8 @@ func (e *DockerComposeExecutor) injectImagePullSecrets(callback EventHandler) in
 		switch s {
 		case api.ImagePullCredentialsStrategyDockerHub:
 			exitCode = e.injectImagePullSecretsForDockerHub(callback, c.EnvVars)
+    case api.ImagePullCredentialsStrategyECR:
+      exitCode = e.injectImagePullSecretsForECR(callback, c.EnvVars)
```
Comment (shiroyasha) [inline pkg/executors/docker_compose_executor.go]: Something is weird with the indentation here. If you are using vim, I recommend vim-go that automatically formats your source code.

Alternatively, you can call go format on the source tree.

File: test/e2e/docker/docker_private_image_ecr.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        {
+          "name": "main",
+          "image": "501965974906.dkr.ecr.us-east-1.amazonaws.com/ecr-playground:latest"
+        }
+      ],
+
+      "image_pull_credentials": [
+        {
+          "env_vars": [
+            { "name": "DOCKER_CREDENTIAL_TYPE", "value": "#{Base64.encode64("ECR")}" },
```
Comment (shiroyasha) [inline test/e2e/docker/docker_private_image_ecr.rb]: Use `AWS_ECR` instead of ECR.

File: pkg/executors/docker_compose_executor.go (hunk 1 above)
Comment (CommanderK5) [reply pkg/executors/docker_compose_executor.go]: Look like my editor is converting <tabs> to <spaces> will run fmt to fix it.

**Comment by shiroyasha** [comment]
Looks good so far. Please do the following too:

- Update the commit messages to more reasonable message and squash them where possible
- Create a test case where the credentials are invalid

**Pull request merged**